src/back/routes/proposal.ts

```diff
@@ -645,33 +645,27 @@ async function getGrants(): Promise<CategorizedGrants> {
   return await ProjectService.getGrants()
 }
 
-// TODO: Still in use by user profile page.
-async function getGrantsByUser(req: Request): ReturnType<typeof getGrants> {
+async function getGrantsByUser(req: Request) {
   const address = validateAddress(req.params.address)
-  const isCoauthoring = req.query.coauthor === 'true'
 
-  let coauthoringProposalIds = new Set<string>()
+  const coauthoring = await CoauthorModel.findProposals(address, CoauthorStatus.APPROVED)
+  const coauthoringProposalIds = new Set(coauthoring.map((coauthoringAttributes) => coauthoringAttributes.proposal_id))
 
-  if (isCoauthoring) {
-    const coauthoring = await CoauthorModel.findProposals(address, CoauthorStatus.APPROVED)
-    coauthoringProposalIds = new Set(coauthoring.map((coauthoringAttributes) => coauthoringAttributes.proposal_id))
-  }
-
-  const grantsResult = await getGrants()
+  const grantsResult = await ProjectService.getProjects()
 
-  const filterGrants = (grants: ProjectWithUpdate[]) => {
-    return grants.filter(
-      (grant) => grant.user.toLowerCase() === address.toLowerCase() || coauthoringProposalIds.has(grant.id)
+  const filterGrants = (projects: ProjectWithUpdate[]) => {
+    return projects.filter(
+      (project) =>
+        project.type === ProposalType.Grant &&
+        (project.user.toLowerCase() === address.toLowerCase() || coauthoringProposalIds.has(project.id))
```

Comment:
You can use `isSameAddress` for this :)

Reply:
I'm thinking we could solve all this in ProposalModel just by querying:

```sql
-- get the authored grants
SELECT * FROM proposals as p
WHERE p.user = '0x549a9021661a85b6bc51c07b3a451135848d0048'
and p.type = 'grant'

UNION

-- get the coauthored grants
SELECT proposals.*
FROM proposals
    JOIN coauthors
        ON proposals.id = coauthors.proposal_id
WHERE
    coauthors.address = '0x549a9021661a85b6bc51c07b3a451135848d0048' AND
    coauthors.status = 'APPROVED' AND
    proposals.type = 'grant';
```

We could add order by date, and then separate them between past and current

Reply:
Agree! I'll create an issue to do this optimization.